docs/core/extensions/logging.md

```diff
@@ -3,34 +3,106 @@ title: Logging in C#
+.NET supports high performance, structured logging via the <xref:Microsoft.Extensions.Logging.ILogger> API to help monitor application behavior and diagnose issues.
+Logs can be written to different destinations by selecting different [logging providers](logging-providers.md). Basic logging providers are built-in and there are many
+others available from third parties.
+
+## Getting started
+
+This first example shows the basics, but it is only suitable for a trivial console app. In the next section we show how to improve the code considering scale, performance, configuration and typical programming patterns.
+
+:::code language="csharp" source="snippets/logging/getting-started/Program.cs" range="1-5":::
+
+The preceding example:
+
+- Creates an <xref:Microsoft.Extensions.Logging.ILoggerFactory>. The ILoggerFactory stores all the configuration that determines where log messages are sent. In this case
+we configured the console [logging provider](logging-providers.md) so that log messages are written to the console.
+- Creates an <xref:Microsoft.Extensions.Logging.ILogger> with a category named `Program`. The [category](#log-category) is a string will that be associated with each message logged
+by this ILogger object. It can be used to group log messages from the same class together when searching or filtering logs.
+- Calls <xref:Microsoft.Extensions.Logging.LoggerExtensions.LogInformation%2A> to log a message at the `Information` level. The [log level](#log-level) indicates the severity of the logged event
+and can be used to filter out less important log messages. The log entry also includes a [message template](#log-message-template) `"Hello World! Logging is {description}."` and a key-value pair
+`description=fun`. The key name comes from the word inside the curly braces in the template and the value comes from the remaining method argument.
 
 [!INCLUDE [logging-samples-browser](includes/logging-samples-browser.md)]
 
-> [!IMPORTANT]
-> Starting with .NET 6, logging services no longer register the <xref:Microsoft.Extensions.Logging.ILogger> type. When using a logger, specify the generic-type alternative <xref:Microsoft.Extensions.Logging.ILogger%601> or register the `ILogger` with [dependency injection (DI)](dependency-injection.md).
+## Logging in a non-trivial app
+
+There are several changes you should consider making to the example above when logging in a less trivial scenario:
+
+- If your application is using [Dependency Injection (DI)](dependency-injection.md) or a host such as ASP.NET's [WebApplication](/aspnet/core/fundamentals/minimal-apis/webapplication) or [Generic Host](generic-host.md) then
+you should use ILoggerFactory and ILogger objects from the DI container rather than creating them directly. See [Integration with DI and Hosts](#integration-with-hosts-and-dependency-injection) below for details.
```

Comment:
When linking to additional information, use the following form:

```md
"For more information, see [link](other-content.md)."
```

```suggestion
- If your application is using [Dependency Injection (DI)](dependency-injection.md) or a host such as ASP.NET's [WebApplication](/aspnet/core/fundamentals/minimal-apis/webapplication) or [Generic Host](generic-host.md) then you should use `ILoggerFactory` and `ILogger` objects from their respective DI containers rather than creating them directly. For more information, see [Integration with DI and Hosts](#integration-with-hosts-and-dependency-injection).
```